src/core/Microsoft.PowerToys.Settings.UI.Runner/MainWindow.xaml.cs

```diff
@@ -20,11 +21,34 @@ private void WindowsXamlHost_ChildChanged(object sender, EventArgs e)
         {
             // Hook up x:Bind source.
             WindowsXamlHost windowsXamlHost = sender as WindowsXamlHost;
-            ShellPage userControl = windowsXamlHost.GetUwpInternalObject() as ShellPage;
+            ShellPage shellPage = windowsXamlHost.GetUwpInternalObject() as ShellPage;
 
-            if (userControl != null)
+            if (shellPage != null)
             {
-                //userControl.XamlIslandMessage = this.WPFMessage;
+                shellPage.SetRestartElevatedCallback(delegate(string msg) 
+                {
+                    MessageBox.Show(
+                        msg,
+                        "Restart Elevated",
+                        MessageBoxButton.OK);
+
+                    Program.ipcmanager.SendMessage(msg);
+
+                    int milliseconds = 2000;
+                    Thread.Sleep(milliseconds);
```

Comment:
Is it possible to redesign this so that we don't have to 'sleep'?  This pattern tends to be either inefficient, or flakey.

Reply:
Sure this is only temporary until we move to .NET Core IPC library which supports threading. Currently suppoting threading and callback functions in ATL is a pain.